### PR TITLE
Convert Imm32 to CsWin32

### PR DESCRIPTION
**Pull request by elachlan**
Relates to #7445

I am unsure if `ImmGetConversionStatus` is correct. So extra eyes on that would be great.

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/winforms/pull/7840)

### CODE REVIEW

File: src/System.Windows.Forms/src/System/Windows/Forms/Control.Ime.cs [hunk 1]
```diff
@@ -1007,30 +1008,35 @@ public static ImeMode GetImeMode(IntPtr handle)
 
             // Determine the IME mode from the conversion status
             Debug.WriteLineIf(CompModSwitches.ImeMode.Level >= TraceLevel.Verbose, "ImmGetConversionStatus(" + inputContext + ", conversion, sentence)");
-            Imm32.ImmGetConversionStatus(inputContext, out Imm32.IME_CMODE conversion, out Imm32.IME_SMODE sentence);
+            IME_CONVERSION_MODE conversion;
+            IME_SENTENCE_MODE sentence;
+            unsafe
+            {
+                PInvoke.ImmGetConversionStatus(inputContext, &conversion, &sentence);
+            }
```
Comment (elachlan) [inline src/System.Windows.Forms/src/System/Windows/Forms/Control.Ime.cs:1016]: I am not 100% on the changes for `ImmGetConversionStatus`

**Review by RussKie — COMMENTED** [review]
LGTM

File: src/System.Windows.Forms.Primitives/src/Windows/Win32/PInvoke.ImmGetContext.cs [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+using Windows.Win32.Globalization;
+
+namespace Windows.Win32
+{
+    internal static partial class PInvoke
+    {
+        public static HIMC ImmGetContext(IHandle<HWND> hWnd)
+        {
+            HIMC result = ImmGetContext(hWnd.Handle);
+            GC.KeepAlive(hWnd);
```
Comment (lonitra) [inline src/System.Windows.Forms.Primitives/src/Windows/Win32/PInvoke.ImmGetContext.cs:14]: We want to use the following existing pattern: 
```suggestion
        public static HIMC ImmGetContext(in T hWnd) where T : IHandle<HWND>
        {
            HIMC result = ImmGetContext(hWnd.Handle);
            GC.KeepAlive(hWnd.Wrapper);
```

File: src/System.Windows.Forms.Primitives/src/Windows/Win32/PInvoke.ImmReleaseContext.cs [hunk 3]
```diff
@@ -2,16 +2,11 @@
...
+    internal static partial class PInvoke
     {
-        [DllImport(Libraries.Imm32)]
-        public static extern BOOL ImmReleaseContext(IntPtr hWnd, IntPtr hIMC);
-
-        public static BOOL ImmReleaseContext(IHandle hWnd, IntPtr hIMC)
+        public static BOOL ImmReleaseContext(IHandle<HWND> hWnd, IntPtr hIMC)
         {
             BOOL result = ImmReleaseContext(hWnd.Handle, hIMC);
             GC.KeepAlive(hWnd);
```
Comment (lonitra) [inline src/System.Windows.Forms.Primitives/src/Windows/Win32/PInvoke.ImmReleaseContext.cs:12]: Same here
```suggestion
        public static BOOL ImmReleaseContext(in T hWnd, HIMC hIMC) where T : IHandle<HWND>
        {
            BOOL result = ImmReleaseContext(hWnd.Handle, hIMC);
            GC.KeepAlive(hWnd.Wrapper);
```

**Review by lonitra — CHANGES_REQUESTED** [review]
Just comments on using existing pattern, but otherwise looks good :)

File: src/System.Windows.Forms.Primitives/src/Windows/Win32/PInvoke.ImmReleaseContext.cs [hunk 4]
```diff
@@ -2,16 +2,13 @@
...
+    internal static partial class PInvoke
     {
-        [DllImport(Libraries.Imm32)]
-        public static extern BOOL ImmReleaseContext(IntPtr hWnd, IntPtr hIMC);
-
-        public static BOOL ImmReleaseContext(IHandle hWnd, IntPtr hIMC)
+        public static BOOL ImmReleaseContext<T>(in T hWnd, HIMC hIMC) where T : IHandle<HWND>
         {
             BOOL result = ImmReleaseContext(hWnd.Handle, hIMC);
             GC.KeepAlive(hWnd);
```
Comment (lonitra) [inline src/System.Windows.Forms.Primitives/src/Windows/Win32/PInvoke.ImmReleaseContext.cs:14]:
```suggestion
            GC.KeepAlive(hWnd.Wrapper);
```

File: src/System.Windows.Forms.Primitives/src/Windows/Win32/PInvoke.ImmGetContext.cs [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+using Windows.Win32.Globalization;
+
+namespace Windows.Win32
+{
+    internal static partial class PInvoke
+    {
+        public static HIMC ImmGetContext<T>(in T hWnd) where T : IHandle<HWND>
+        {
+            HIMC result = ImmGetContext(hWnd.Handle);
+            GC.KeepAlive(hWnd);
```
Comment (lonitra) [inline src/System.Windows.Forms.Primitives/src/Windows/Win32/PInvoke.ImmGetContext.cs:14]:
```suggestion
            GC.KeepAlive(hWnd.Wrapper);
```

**Review by lonitra — APPROVED** [review]
LGTM, thank you!

**Pull request merged**